velox/connectors/Connector.h

```diff
@@ -226,6 +226,21 @@ class DataSource {
   virtual int64_t estimatedRowSize() {
     return kUnknownRowSize;
   }
+
+  // Returns the filter for the column identified by @param outputChannel.
+  virtual std::shared_ptr<common::Filter> getColumnFilter(
+      column_index_t outputChannel) {
+    VELOX_UNSUPPORTED("getFilter");
+  }
+
+  // Sets the filter to the column identified by @param outputChannel.
+  // This can be called when passing the column filters to a preloading data
+  // source. The purpose of this is to avoid the overhead of filters merging.
+  virtual void setColumnFilter(
```

Comment:
How is this different from addDynamicFilter? Having these two APIs is confusing. Please, update PR description to provide overall context for this change.

Reply:
@zhli1142015 Thank you for this additional context. Is there a GitHub issue about this? Sounds like we need to iterate on the design a bit and it will be easier to do in a GitHub issue.